pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -62,6 +62,11 @@ func (h *AnnotationBackend) Record(ctx context.Context, rule history_model.RuleM
 	panel := parsePanelKey(rule, logger)
 
 	errCh := make(chan error, 1)
+	if len(annotations) == 0 {
```

Comment:
My only concern is that there's no distinction between recording and not recording to the caller of this function - but maybe this is captures by metrics and therefore is fine?

Reply:
That's how it already works to begin with. Right now the scheduler and state packages call Record on **every evaluation**, and they pass in in tons of states that we don't actually care about or actually want to record, like Alerting->Alerting. Regardless of this change, it relies on getting called a _ton_ and nothing in the caller cares if it recorded or didn't.